naming/src/main/java/com/alibaba/nacos/naming/consistency/ephemeral/distro/v2/DistroClientDataProcessor.java

```diff
@@ -166,16 +168,35 @@ private void upgradeClient(Client client, ClientSyncData clientSyncData) {
         List<String> groupNames = clientSyncData.getGroupNames();
         List<String> serviceNames = clientSyncData.getServiceNames();
         List<InstancePublishInfo> instances = clientSyncData.getInstancePublishInfos();
+        List<BatchInstancePublishInfo> batchInstancePublishInfos = clientSyncData.getBatchInstancePublishInfos();
         Set<Service> syncedService = new HashSet<>();
         for (int i = 0; i < namespaces.size(); i++) {
             Service service = Service.newService(namespaces.get(i), groupNames.get(i), serviceNames.get(i));
             Service singleton = ServiceManager.getInstance().getSingleton(service);
             syncedService.add(singleton);
-            InstancePublishInfo instancePublishInfo = instances.get(i);
-            if (!instancePublishInfo.equals(client.getInstancePublishInfo(singleton))) {
-                client.addServiceInstance(singleton, instancePublishInfo);
-                NotifyCenter.publishEvent(
-                        new ClientOperationEvent.ClientRegisterServiceEvent(singleton, client.getClientId()));
+            InstancePublishInfo clientInstancePublishInfo = client.getInstancePublishInfo(singleton);
+            if (CollectionUtils.isNotEmpty(batchInstancePublishInfos)) {
+                BatchInstancePublishInfo batchInstancePublishInfo = batchInstancePublishInfos.get(i);
+                List<InstancePublishInfo> sourceInstanceInfo = batchInstancePublishInfo.getInstancePublishInfos();
+                BatchInstancePublishInfo targetInstanceInfo = (BatchInstancePublishInfo) clientInstancePublishInfo;
+                boolean result = false;
+                //If it is not empty, compare whether the instances are the same. If it is empty, notify directly
+                if (targetInstanceInfo != null) {
+                    result  = CollectionUtils.isEqualCollection(sourceInstanceInfo,
```

Comment:
Why not override equals and hashcode?

Reply:
This method can achieve the same purpose. It is also used to compare whether the fields of the internal elements of the set are equal. It has been tested

Reply:
> Why not override equals and hashcode?

This method can achieve the same purpose. It is also used to compare whether the fields of the internal elements of the set are equal. It has been tested

Reply:
> Why not override equals and hashcode?

i fix it